src/psyclone/tests/gocean1p0_stencil_test.py

```diff
@@ -0,0 +1,227 @@
+# GOStencil is provided with invalid stencil information
+
+
+def test_stencil_invalid_format():
+    '''Check all the ways in which the stencil information can be invalid'''
+    stencil = GOStencil()
+
+    # this should work as it is valid
+    stencil_string = "stencil(000,011,000)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    stencil.load(parsed_stencil, "kernel_stencil")
+
+    # this should cause a general unexpected format error
+    stencil_string = "(a)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert "expected either a name or the format 'stencil(...)" \
+        in str(excinfo.value)
+
+    # this should cause an unsupported name error
+    stencil_string = "random_name"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert ("argument is 'random_name' but must be one of ['pointwise'] or "
+            "stencil(...)") in str(excinfo.value)
+
+    # this should cause an unsupported name error
+    stencil_string = "stenci(a)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert "argument is 'stenci' but must be 'stencil(...)" \
+        in str(excinfo.value)
+
+    # this should cause a not-enough-args error
+    stencil_string = "stencil(a)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert "format 'stencil(...)', has 1 arguments but should have 3" \
+        in str(excinfo.value)
+
+    # this should cause a not-a-number error
+    stencil_string = "stencil(a,b,c)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert "Argument index 0 should be a number but found 'a'" \
+        in str(excinfo.value)
+
+    # this should also cause a not-a-number error
+    stencil_string = "stencil(000,x00,000)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert "index 1 should be a number but found 'x00'" \
+        in str(excinfo.value)
+
+    # this should cause a not-3-numbers error
+    stencil_string = "stencil(012,345,67)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert "index 2 should consist of 3 numbers but found 2" \
```

Comment:
This error message should say "digits" instead of "numbers".

Reply:
Agreed. Done.